backend/test/job_api_test.go

```diff
@@ -1 +1,308 @@
+	listJobsResponse, err = s.jobClient.ListJobs(ctx, &api.ListJobsRequest{PageSize: 1})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listJobsResponse.Jobs))
+	assert.Equal(t, "hello world", listJobsResponse.Jobs[0].Name)
+	listJobsResponse, err = s.jobClient.ListJobs(ctx, &api.ListJobsRequest{PageSize: 1, PageToken: listJobsResponse.NextPageToken})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listJobsResponse.Jobs))
+	assert.Equal(t, "argument parameter", listJobsResponse.Jobs[0].Name)
+
+	/* ---------- List the jobs, paginated, sort by name ---------- */
+	listJobsResponse, err = s.jobClient.ListJobs(ctx, &api.ListJobsRequest{PageSize: 1, SortBy: "name"})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listJobsResponse.Jobs))
+	assert.Equal(t, "argument parameter", listJobsResponse.Jobs[0].Name)
+	listJobsResponse, err = s.jobClient.ListJobs(ctx, &api.ListJobsRequest{PageSize: 1, SortBy: "name", PageToken: listJobsResponse.NextPageToken})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listJobsResponse.Jobs))
+	assert.Equal(t, "hello world", listJobsResponse.Jobs[0].Name)
+
+	/* ---------- List the jobs, sort by unsupported field ---------- */
+	_, err = s.jobClient.ListJobs(ctx, &api.ListJobsRequest{PageSize: 2, SortBy: "description"})
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "InvalidArgument")
+
+	/* ---------- List jobs for hello world experiment. One job should be returned ---------- */
+	listJobsResponse, err = s.jobClient.ListJobs(ctx, &api.ListJobsRequest{
+		ResourceReferenceKey: &api.ResourceKey{
+			Type: api.ResourceType_EXPERIMENT, Id: helloWorldExperiment.Id}})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listJobsResponse.Jobs))
+	assert.Equal(t, "hello world", listJobsResponse.Jobs[0].Name)
+
+	// The scheduledWorkflow CRD would create the run and it synced to the DB by persistent agent.
+	// This could take a few seconds to finish.
+	// TODO: Retry list run every 5 seconds instead of sleeping for 40 seconds.
+	time.Sleep(40 * time.Second)
+
+	/* ---------- Check run for hello world job ---------- */
+	listRunsResponse, err := s.runClient.ListRuns(ctx, &api.ListRunsRequest{
+		ResourceReferenceKey: &api.ResourceKey{
+			Type: api.ResourceType_EXPERIMENT, Id: helloWorldExperiment.Id}})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listRunsResponse.Runs))
+	helloWorldRun := listRunsResponse.Runs[0]
+	s.checkHelloWorldRun(t, helloWorldRun, helloWorldExperiment.Id, helloWorldJob.Id, requestStartTime)
+
+	/* ---------- Check run for argument parameter job ---------- */
+	listRunsResponse, err = s.runClient.ListRuns(ctx, &api.ListRunsRequest{
+		ResourceReferenceKey: &api.ResourceKey{
+			Type: api.ResourceType_EXPERIMENT, Id: argParamsExperiment.Id}})
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(listRunsResponse.Runs))
+	argParamsRun := listRunsResponse.Runs[0]
+	s.checkArgParamsRun(t, argParamsRun, argParamsExperiment.Id, argParamsJob.Id, requestStartTime)
+
+	/* ---------- Clean up ---------- */
```

Comment:
Could we please turn this into a utility that anyone can reuse? Otherwise all the e2e tests have to do their own clean up in the current setup. 

Additionally, since these tests are not concurrent, let's use the list calls to list and then delete all the resources.